pkg/instrumentation/kafka/option.go

```diff
@@ -0,0 +1,72 @@
+package kafka
+
+import (
+	"math"
+	"os"
+	"strconv"
+)
+
+type config struct {
+	consumerServiceName string
+	producerServiceName string
+	analyticsRate       float64
+}
+
+// An Option customizes the config.
+type Option func(cfg *config)
+
+func newConfig(opts ...Option) *config {
+	cfg := &config{
+		consumerServiceName: "kafka",
+		producerServiceName: "kafka",
+		analyticsRate:       math.NaN(),
+	}
+
+	var datadogAnalyticsEnabled bool
+	datadogAnalyticsEnabledString := os.Getenv("DD_TRACE_KAFKA_ANALYTICS_ENABLED")
```

Comment:
Should we document the ENV variable in the README?

I believe this is borrowed from [`confluent-kafka-go`](https://github.com/DataDog/dd-trace-go/blob/d57ebc3f216cec67a9baa3f5f57dd44f3978df0b/contrib/confluentinc/confluent-kafka-go/kafka/option.go#L34) DD tracer, right?

Reply:
@fotos well, this code ported here from other tracers. I don't think we have usage of trace analytics atm. Probably, I'd keep it an undocumented feature and encourage using the `WithAnalytics` option.